src/servers/http_server.cc

```diff
@@ -553,10 +566,27 @@ HTTPServerImpl::InferRequest::FinalizeResponse()
 
 Status
 HTTPServer::Create(
-    InferenceServer* server, uint16_t port, int thread_cnt,
-    std::unique_ptr<HTTPServer>* http_server)
+    InferenceServer* server,
+    std::map<int32_t, std::vector<std::string>> port_map, int thread_cnt,
+    std::vector<std::unique_ptr<HTTPServer>>* http_servers)
 {
-  http_server->reset(new HTTPServerImpl(server, port, thread_cnt));
+  size_t i = 0;
+
+  if (!port_map.empty()) {
+    for (auto const& ep_map : port_map) {
+      std::string addr = "0.0.0.0:" + std::to_string(ep_map.first);
+      LOG_INFO << "Starting HTTPService at " << addr;
+      ((*http_servers)[i])
```

Comment:
Same comment as above, don't require caller to pre-allocator http_servers. Clear it and push_back.